File: University Student Rating/Code/androidclient/UniversityStudentRating/app/src/main/java/com/example/arturrinkis/universitystudentrating/ServerServiceAPI/Interfaces/Services/IDataService.java
package com.example.arturrinkis.universitystudentrating.ServerServiceAPI.Interfaces.Services;

import android.graphics.Bitmap;

import com.example.arturrinkis.universitystudentrating.DTO.BranchProfessorDiscipline;
import com.example.arturrinkis.universitystudentrating.DTO.City;
import com.example.arturrinkis.universitystudentrating.DTO.Country;
import com.example.arturrinkis.universitystudentrating.DTO.Course;
import com.example.arturrinkis.universitystudentrating.DTO.Discipline;
import com.example.arturrinkis.universitystudentrating.DTO.Faculty;
import com.example.arturrinkis.universitystudentrating.DTO.GenderType;
import com.example.arturrinkis.universitystudentrating.DTO.ProfessorDiscipline;
import com.example.arturrinkis.universitystudentrating.DTO.RatingDivision;
import com.example.arturrinkis.universitystudentrating.DTO.Status;
import com.example.arturrinkis.universitystudentrating.DTO.StudentProfessorDiscipline;
import com.example.arturrinkis.universitystudentrating.DTO.University;

import java.util.ArrayList;

public interface IDataService {
    String GetServiceURL();
    ArrayList<GenderType> getGenderTypes();
    ArrayList<Country> getCountries();
    ArrayList<City> getCitiesByCountry(int countryId);
    ArrayList<University> getUniversitiesByCity(int cityId);
    ArrayList<Status> getStatuses();
    ArrayList<Faculty> getFacultiesByUniversity(int universityId);
    ArrayList<Course> getCourses();
    ArrayList<ProfessorDiscipline> getProfessorDisciplines(int professorId);
    ArrayList<Discipline> getDisciplinesByProfessor(int professorId);
    ArrayList<StudentProfessorDiscipline> getStudentProfessorDisciplines(int studentId);
    BranchProfessorDiscipline getBranchProfessorDiscipline(int studentId, int universityId);
    ArrayList<RatingDivision> GetRatingDivisions();
    boolean uploadPhotoImageFile(Bitmap bitmap, String filePath);
}

File: University Student Rating/Code/server/UniversityStudentRatingService/ServiceLogicLayer/Interfaces/IDataService.cs
using DataAccessLayer.DbContext;
using ServiceLogicLayer.DTO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceLogicLayer.Interfaces
{
    public interface IDataService : IDisposable
    {
        IEnumerable<GenderType> GetGenderTypes();
        IEnumerable<Country> GetCountries();
        IEnumerable<City> GetCitiesByCountry(int countryId);
        IEnumerable<University> GetUniversitiesByCity(int cityId);
        IEnumerable<Status> GetStatuses();
        IEnumerable<Faculty> GetFacultiesByUniversity(int universityId);
        IEnumerable<Course> GetCourses();
        IEnumerable<ProfessorDiscipline> GetProfessorDisciplines(int branchId, int universityId);
        IEnumerable<ProfessorDiscipline> GetProfessorDisciplines(int professorId);
        IEnumerable<Discipline> GetDisciplinesByProfessor(int professorId);
        IEnumerable<StudentProfessorDiscipline> GetStudentProfessorDisciplines(int studentId);
        BranchProfessorDisciplineDTO GetBranchProfessorDiscipline(int studentId, int universityId);
        IEnumerable<RatingDivision> GetRatingDivisions();
    }
}